### PR TITLE
[14.0] Add requests_operating_unit

### PR DESCRIPTION
**Pull request by flook-chaiwat**
Add new module - requests_operating_unit

### CODE REVIEW

File: requests_operating_unit/models/request.py [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+        return res
+
+    def write(self, vals):
+        self.clear_caches()
+        return super(OperatingUnit, self).write(vals)
+
+
+class RequestOperatingUnit(models.Model):
+
+    _inherit = "request.request"
```
Comment (kittiu) [inline requests_operating_unit/models/request.py:75]:
```suggestion
    _inherit = "request.request"
   
```

File: requests_operating_unit/models/res_users.py [hunk 2]
```diff
@@ -0,0 +1,72 @@
+# Copyright 2015-TODAY ForgeFlow
+# - Jordi Ballester Alomar
+# Copyright 2015-TODAY Serpent Consulting Services Pvt. Ltd. - Sudhir Arya
+# License LGPL-3.0 or later (https://www.gnu.org/licenses/lgpl.html).
```
Comment (kittiu) [inline requests_operating_unit/models/res_users.py]: Please change all files, to be Ecosoft (please see sample from i.e., OCA/l10n-thailand)

File: requests_operating_unit/models/request.py [hunk 3]
```diff
@@ -0,0 +1,81 @@
+# Copyright 2015-TODAY Eficent
+# - Jordi Ballester Alomar
+# Copyright 2015-TODAY Serpent Consulting Services Pvt. Ltd. - Sudhir Arya
+# License LGPL-3.0 or later (https://www.gnu.org/licenses/lgpl.html).
+from odoo import api, fields, models
+
+
+class OperatingUnit(models.Model):
```
Comment (kittiu) [inline requests_operating_unit/models/request.py]: We don't need this model. We just inherit module, operating_unit in __manifest__.py and we are good.

File: requests_operating_unit/models/res_users.py [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+from odoo import api, fields, models
+
+
+class ResUsers(models.Model):
```
Comment (kittiu) [inline requests_operating_unit/models/res_users.py]: We don't need this.

File: requests_operating_unit/security/ir.model.access.csv [hunk 5]
```diff
@@ -0,0 +1,3 @@
+id,name,model_id/id,group_id/id,perm_read,perm_write,perm_create,perm_unlink
+access_operating_unit_manager,operating_unit_manager,model_operating_unit,group_manager_operating_unit,1,1,1,1
+access_operating_unit_user,operating_unit_user,model_operating_unit,group_multi_operating_unit,1,0,0,0
```
Comment (kittiu) [inline requests_operating_unit/security/ir.model.access.csv]: We don't need these

File: requests_operating_unit/__manifest__.py [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+{
+    "name": "Requests_Operating_Unit",
+    "version": "14.0.1.0.0",
+    "author": "Ecosoft, Odoo Community Association (OCA)",
+    "category": "Human Resources/Requests",
+    "website": "https://github.com/OCA/requests",
+    "license": "AGPL-3",
+    "summary": "Create and validate requests",
+    "depends": ["mail", "hr", "product", "requests"],
```
Comment (kittiu) [inline requests_operating_unit/__manifest__.py]:
```suggestion
    "depends": ["requests", "operating_unit"],
```

File: requests_operating_unit/data/operating_unit_data.xml [hunk 7]
```diff
@@ -0,0 +1,17 @@
+<?xml version="1.0" encoding="utf-8" ?>
+<odoo noupdate="1">
+    <!-- Basic operating Unit  -->
+    <record id="main_operating_unit" model="operating.unit">
```
Comment (kittiu) [inline requests_operating_unit/data/operating_unit_data.xml]: We don't need all theses.

File: requests_operating_unit/models/request.py [hunk 8]
```diff
@@ -0,0 +1,14 @@
+# Copyright 2021 Ecosoft
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+from odoo import fields, models
+
+
+class RequestOperatingUnit(models.Model):
+
+    _inherit = "request.request"
+    operating_unit_id = fields.Many2one(
```
Comment (kittiu) [inline requests_operating_unit/models/request.py]: I think this should be spacing like this, nicer.

```
class RequestOperatingUnit(models.Model):
    _inherit = "request.request"

    operating_unit_id = fields.Many2one(
    ....
```

File: requests_operating_unit/readme/CONTRIBUTORS.rst [hunk 9]
```diff
@@ -0,0 +1,2 @@
+* Kitti U. <kittiu@ecosoft.co.th>
```
Comment (kittiu) [inline requests_operating_unit/readme/CONTRIBUTORS.rst]:
```suggestion
```

File: requests_operating_unit/security/operating_unit_security.xml [hunk 10]
```diff
@@ -0,0 +1,22 @@
...
+<odoo noupdate="1">
+
+    <record id="ir_rule_request_operating_units" model="ir.rule">
+        <field name="model_id" ref="requests.model_request_request" />
+        <field name="domain_force">
+            ['|','|',('operating_unit_id','=',False),
+            ('operating_unit_id','=',user.default_operating_unit_id.id),
+            ('operating_unit_id','in',[g.id for g in user.operating_unit_ids])]
+        </field>
+        <field name="name">Requests from allowed operating units</field>
```
Comment (kittiu) [inline requests_operating_unit/security/operating_unit_security.xml]:
```suggestion
        <field name="name">Requests form allowed operating units</field>
```

**Comment by kittiu** [comment]
@flook-chaiwat I think in the future PR, you don't need to fork. You can use ecosoft-odoo, just make a branch. This way, if you are not around, other people in ecosoft can take care your code too.

**Comment by flook-chaiwat** [comment]
@kittiu I have added constraints for approvers, they must be in the same operating unit.
`@api.constrains("approver_ids", "operating_unit_id")
    def _check_approver_operating_unit(self):
        for r in self.approver_ids:
            if (
                self.operating_unit_id
                not in r.user_id.sudo().operating_unit_ids
            ):
                raise exceptions.ValidationError(`

**Pull request closed without merging**